docs/user_guide/secure_payloads.rst

```diff
@@ -0,0 +1,233 @@
+large your payload is, you may need to increase the size of this mounted
+partition by adjusting the `secure_size` option in `keylime.conf`.
+
+This simple mode of operation is suitable for many situations where the secrets
+and other bootstrapping information is basic.  However, there are several
+features that Keylime supports like revocation and certificate management that
+do not work in this mode.  For those, you'll need the next mode: Certificate
+Package Mode.
+
+
+Certificate Package Mode
+------------------------
+
+This mode of Keylime automates many common actions that tenants will want to do
+when provisioning their Agents.  First, Keylime can create an X509 certificate
+authority (CA) and then issue certificates and corresponding private keys to each
+provisioned node.  This CA lives on the tenant and can be used to bootstrap many
+other security solutions like mutual TLS or SSH.  To use this mode, pass
+the `--cert` option and a directory where the CA is located as the parameter to
+this option.  Keylime will then create a certificate for the node (with the common
+name set to the Agent's UUID) and then create a zip file containing the newly
+generated X509 certificates, trust roots, and private keys. It then uses the
+same process for single file encryption as described above to securely deliver
+all the keys to the Agent.  Optionally, the user can specify with the `--include`
+option a directory of additional files to be put into the certification package zip
+and securely delivered to the Agent.
+
+This mode of operation also natively supports certificate revocation. If the Keylime
+Verifier detects an Agent that no longer satisfies its integrity policy (e.g., it booted
+an authorized kernel or ran an unauthorized binary not on the IMA whitelist), it
+will create a signed revocation notification.  These revocation notifications are
+signed by a special certificate/private key called the RevocationNotifier.  Keylime
+will automatically create this certificate and pass it to the verifier when you add
+a new Agent to the verifier.  Keylime will also include the public certificate for
+this key in the zip it sends to the Agent. This way Agents can validate the
+revocation notifications they receive from the verifier.
+
+By default all Keylime Agents listen for these revocation notifications (see
+the `listen_notifications` option in `keylime.conf`). Using the keys in the
+unzipped certificate package, Agents can check that the revocations are valid.
+Keylime Agents can also take actions in response to a valid revocation.
+You can configure these actions by putting additional files into the delivered zip
+file using `--include`.
+
+Revocation actions are small Python scripts that will run on an Agent when a valid
+revocation is received.  They should contain an `execute` function that takes
+one argument.  This argument is a Python dictionary of metadata that can be used
```

Comment:
It would be nice to have this Python dictionary documented somewhere. When I was working on revocation actions I had to first create a dummy action that just dumped the contents of the dict so I could figure out how to use it.

Reply:
Absolutely, tough this sounds like something a bit out of scope for this document / PR.
I think we should create an issue to track this.

Reply:
I think it should actually live in this documented (unless there's a better place for it that I'm unaware of). But it doesn't have to be part of this current PR and can just be a future improvement.

Reply:
Tracking this in https://github.com/keylime/keylime-docs/issues/54. Let's keep the conversation going there.